python/cudf/cudf/core/column/column.py

```diff
@@ -1399,6 +1415,19 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                 size=pa_size,
                 offset=pa_offset,
             )
+        elif isinstance(arbitrary, pa.DurationArray):
```

Comment:
Could we move this to a `Column.from_arrow()` constructor? https://github.com/rapidsai/cudf/issues/5697

Reply:
Made this refactor for all the columns